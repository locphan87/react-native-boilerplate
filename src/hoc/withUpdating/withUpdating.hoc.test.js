import { View } from 'react-native'
import { isFunction } from 'ramda-adjunct'

import { testSnapshots } from '../../utils/test.util'

import withUpdatingConfig, {
  simulatePending,
  withUpdating
} from './withUpdating.hoc'

jest.mock('recompose', () => {
  const origin = require.requireActual('recompose')

  return {
    ...origin,
    withState: stateName => `withState-${stateName}`,
    withProps: getProps => {
      const newProps = getProps({ createGoal: jest.fn() })
      const propKeys = Object.keys(newProps).toString()

      return `withProps-[${propKeys}]`
    },
    compose: (...args) => args
  }
})

describe('withUpdating', () => {
  const Component = withUpdating(View)
  testSnapshots(Component, [
    {
      props: { updating: true },
      description: 'updating state'
    },
    {
      props: { updating: false },
      description: 'normal state'
    }
  ])
})

describe('withUpdatingConfig', () => {
  test('should return expected list of HOCs', () => {
    const actual = withUpdatingConfig(['createGoal'])
    const expected = [
      'withState-updating',
      'withProps-[createGoal]',
      withUpdating
    ]
    expect(actual).toEqual(expected)
  })
})

describe('simulatePending', () => {
  const updates = ['getArticleList']
  const props = {
    setUpdating: jest.fn(),
    getArticleList: jest.fn(() => Promise.resolve([]))
  }
  test('should call setUpdating before each update on SUCESS', async () => {
    const actual = simulatePending(updates)(props)
    expect(isFunction(actual.getArticleList)).toBe(true)
    await actual.getArticleList()
    expect(props.setUpdating).toBeCalled()
  })
})
