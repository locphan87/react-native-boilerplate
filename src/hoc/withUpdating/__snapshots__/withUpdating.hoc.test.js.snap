// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUpdating snapshots normal state 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
</View>
`;

exports[`withUpdating snapshots updating state 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
  <LoadingMask
    style={
      Object {
        "backgroundColor": "rgba(31, 32, 39, 1)",
        "opacity": 0.8,
      }
    }
  />
</View>
`;
